### PR TITLE
merge partybuy into partybid

### PR DESCRIPTION
**Pull request by msfeldstein**
PartyBid and PartyBuy share a lot of code and it would be cleaner to have them in the same repo.
PartyBid.sol and PartyBuy.sol both inherit from Party.sol which is identical
Lots of helper methods and utilities for deployment and testing

This brings in all the code from the current main branch of partybuy, but it does not include annas new CollectionBuy changes, so we'll need to make sure to get that in.

**This should be reviewed commit by commit**; the really big commits are just shuffling files around so probably don't need to be looked at much.

It also bumps solidity version to suppress some warnings that OZ contracts were throwing and cleans up some other odds and ends.

There is more unification that can be done but this is a good start.

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -15,16 +15,7 @@ module.exports = {
...
-        settings: {
-          optimizer: {
-            enabled: true,
-            runs: 10000,
-          },
-        },
-      },
-      {
-        version: '0.8.5',
+        version: '0.8.11',
```
Comment (anna-carroll) [inline hardhat.config.js]: no optimizer? reasoning?

File: package.json [hunk 2]
```diff
@@ -5,8 +5,8 @@
   "main": "",
   "scripts": {
     "test": "npm run compile && hardhat test",
-    "deploy": "npm run compile && node ./deploy/index.js && npm run verify",
-    "deploy-partybuy": "npm run compile && node ./deploy-partybuy/index.js && npm run verify-partybuy",
+    "deploy:partybid": "npm run compile && node ./deploy/partybid/index.js && npm run verify",
+    "deploy:partybuy": "npm run compile && node ./deploy/partybuy/index.js && npm run verify-partybuy",
     "verify": "hardhat --network rinkeby verify-contracts",
```
Comment (anna-carroll) [inline package.json:10]: change `verify:partybid` and `verify:partybuy`

**Review by anna-carroll — COMMENTED** [review]
overall looking good to me

**Review by msfeldstein — COMMENTED** [review]
addressed comments:
- uniform names for tasks
- uniform folder structure in tests

File: deploy/partybid/deployed-contracts/rinkeby.json [hunk 3]
```diff
@@ -6,5 +6,8 @@
     "foundation": "0x3183F5AcD2BCD82a4587aA4485DF92Fee3c9bA8b",
     "zora": "0x8bdf23481fcB9e640F14F24030eF7E273D100643",
     "nouns": "0xB304bBf823CeD9c7a41a8747ec017C5F42d81105"
-  }
+  },
+  "partyBuyFactory": "0x554d474D9A3cDb6185A064b376B7031B2B66c401",
```
Comment (anna-carroll) [inline deploy/partybid/deployed-contracts/rinkeby.json:10]: partybuy addresses in partybid folder

File: test/custom/AuctionCanceled.test.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ const {
 } = require('../helpers/externalTransactions');
 const { deployTestContractSetup } = require('../helpers/deploy');
 const { MARKETS, MARKET_NAMES, PARTY_STATUS } = require('../helpers/constants');
-const { testCases } = require('../testCases.json');
+const { testCases } = require('../partyBidTestCases.json');
```
Comment (anna-carroll) [inline test/custom/AuctionCanceled.test.js:18]: nit: folders are `partybuy` elsewhere, but `PartyBuy` in tests

File: test/custom/TokenGating.test.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ const {
 const { placeBid } = require('../helpers/externalTransactions');
 const { deployTestContractSetup, deploy, getTokenVault } = require('../helpers/deploy');
 const { MARKETS, FOURTY_EIGHT_HOURS_IN_SECONDS } = require('../helpers/constants');
-const { testCases } = require('../testCases.json');
+const { testCases } = require('../partyBidTestCases.json');
```
Comment (anna-carroll) [inline test/custom/TokenGating.test.js:19]: partybuy tests are in PartyBuy folder, but partybid tests are in root folder - should both be in sub-folders?

**Review by anna-carroll — APPROVED** [review]
no blocking comments!

**Pull request merged**